restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -589,6 +595,65 @@ private void invite(final Object message) throws IOException, NumberParseExcepti
 
     }
 
+    /**
+     * FIXME: duplicated code make into static function or something more optimized
+     * Replace headers
+     * @param SipServletRequest message
+     * @param Map<String, ArrayList<String> > headers
+     */
+    private void addHeadersToMessage(SipServletRequest message, Map<String, ArrayList<String> > headers) {
```

Comment:
@gvagenas @abdulazizali77 if we want to restrict headers we should do that at API level when they are first provided in the extension so we do it only once and not deep in the code everytime.